### DISCUSSION TITLE
Upgrading from 2.15.2.Final to 2.16.4.Final breaks custom build profiles

### DISCUSSION THREAD

**Discussion by ghubstan** [quarkusio/quarkus#31657] [Q&A] [answered]
I have two custom build profiles in my `application.properties` file.  _Note: profile names are kebab-case._

I have a profile to build and push a jar image, i.e.,
`%k8s-jib-jar.quarkus.package.type=jar`
...

I have a profile to build and push a native image, i.e.,
`%k8s-jib-native.quarkus.package.type=native`
...

I use one or the other custom build profile via the gradle build command:

`./gradlew clean build -Dquarkus-profile=k8s-jib-jar `
OR
`./gradlew clean build -Dquarkus-profile=k8s-jib-native`

My custom profile properties work fine when using `2.15.2.Final`, but are ignored when using `2.16.4.Final`.

I cannot find anything (yet) in the Quarkus migration guide or github issues that indicates a change in the way custom profiles are defined and used at build-time.

Can anyone advise?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
@radcortez, any ideas?

**Reply by radcortez** [reply]
I'll have a look

**Comment by radcortez** [comment, marked as answer]
@ghubstan please use `quarkus.profile` (with a dot separator) instead of a dash. The `quarkus-profile` has been deprecated a long time ago (in 2019), but only now got removed.

We are sorry for the inconvenience and for not property communicating that `quarkus-profile` should not be used.

**Reply by ghubstan** [reply]
> @ghubstan please use `quarkus.profile` (with a dot separator) instead of a dash. The `quarkus-profile` has been deprecated a long time ago (in 2019), but only now got removed.
> 
> We are sorry for the inconvenience and for not property communicating that `quarkus-profile` should not be used.

I changed my build command(s) to use `quarkus.profile` -- to determine with profile to build, e.g.,

         ./gradlew clean build -Dquarkus.profile=k8s-jib-native  -Dquarkus.container-image.push=true

Now,  gradle/quarkus uses my chosen profile.  Thanks.

It seems to work fine for building a jar image, but fails with a permissions error when building a native img.  
See new topic:  https://github.com/quarkusio/quarkus/discussions/31696
